### DISCUSSION TITLE
PingFederate + First Adapter

### DISCUSSION THREAD

**Discussion by jsolares-nfl** [Versent/saml2aws#1426] [Ideas and Feature Requests]
First adapter is basically a form in pingfederate asking for an email address, so you can route authentication depending on the domain, it can be sent through to pingfederate+pingid authentication or to a different federated idp provider. the following works if it goes on through to pingfederate + 2fa

[pingfed.go.zip](https://github.com/user-attachments/files/17050863/pingfed.go.zip)

DEBU[0009] HTTP Req                                      URL="https://pingfed.test.com/idp/startSSO.ping?PartnerSpId=urn:amazon:webservices" http=client method=GET
DEBU[0009] HTTP Res                                      Status="200 OK" http=client
DEBU[0009] doc detect                                    provider=pingfed type="first form adapter"
DEBU[0009] HTTP Req                                      URL="https://pingfed.test.com/idp/N9JXUjx2mo/resumeSAML20/idp/startSSO.ping" http=client method=POST
DEBU[0009] HTTP Res                                      Status="200 OK" http=client
DEBU[0009] doc detect                                    provider=pingfed type=login
DEBU[0009] HTTP Req                                      URL="https://pingfed.test.com/idp/N9JXUjx2mo/resumeSAML20/idp/startSSO.ping" http=client method=POST
DEBU[0011] HTTP Res                                      Status="200 OK" http=client
DEBU[0011] doc detect                                    provider=pingfed type=form-redirect
DEBU[0011] HTTP Req                                      URL="https://authenticator.pingone.com/pingid/ppm/auth" http=client method=POST
DEBU[0012] HTTP Res                                      Status="200 " http=client
DEBU[0012] doc detect                                    provider=pingfed type=webauthn
DEBU[0012] HTTP Req                                      URL="https://authenticator.pingone.com/pingid/ppm/auth" http=client method=POST
DEBU[0012] HTTP Res                                      Status="200 " http=client
DEBU[0012] doc detect                                    provider=pingfed type=swipe
DEBU[0015] HTTP Req                                      URL="https://authenticator.pingone.com/pingid/ppm/auth/status" http=client method=GET
DEBU[0015] HTTP Res                                      Status="200 " http=client
DEBU[0018] HTTP Req                                      URL="https://authenticator.pingone.com/pingid/ppm/auth/status" http=client method=GET
DEBU[0018] HTTP Res                                      Status="200 " http=client
DEBU[0021] HTTP Req                                      URL="https://authenticator.pingone.com/pingid/ppm/auth/status" http=client method=GET
DEBU[0022] HTTP Res                                      Status="200 " http=client
DEBU[0022] HTTP Req                                      URL="https://authenticator.pingone.com/pingid/ppm/auth/response" http=client method=GET
DEBU[0022] HTTP Res                                      Status="200 " http=client
DEBU[0022] doc detect                                    provider=pingfed type=form-redirect
DEBU[0022] HTTP Req                                      URL="https://pingfed.test.com/idp/N9JXUjx2mo/resumeSAML20/idp/startSSO.ping" http=client method=POST
DEBU[0023] HTTP Res                                      Status="200 OK" http=client
DEBU[0023] doc detect                                    provider=pingfed type=saml-response-to-aws

**Comment by RossRichardson** [comment]
Hi @jsolares-nfl, will accept Pull Requests if you are willing to use that process. Will treat this as a feature request and move to ideas in the discussions area until a well formed issue for a feature, and pull request, exists.

**Reply by jsolares-nfl** [reply]
I've added https://github.com/Versent/saml2aws/pull/1427

**Reply by jasons42** [reply]
@RossRichardson Is there anything else needed to get #1427 merged? Would love to have this feature in our org, our current authentication process is pretty annoying.

**Reply by bsamsom** [reply]
@tinaboyce i see you've merged/approved some other prs, as this has sat without feedback/approval for a year trying to ping someone else to look at it.
